pkg/audit/verifier.go

```diff
@@ -489,24 +517,45 @@ func (verifier *Verifier) GetShare(ctx context.Context, limit *pb.AddressedOrder
 	}, nil
 }
 
+// RemoveFailedPieces removes lost pieces from a pointer
+func (verifier *Verifier) RemoveFailedPieces(ctx context.Context, path string, pointer *pb.Pointer, failedPieces storj.NodeIDList) (err error) {
+	defer mon.Task()(&ctx)(&err)
+
+	if len(path) == 0 {
+		return nil
+	}
+	remoteSegment := pointer.GetRemote()
+	for i, piece := range remoteSegment.GetRemotePieces() {
+		for _, failedPiece := range failedPieces {
+			if piece.NodeId == failedPiece {
+				remoteSegment.RemotePieces = append(remoteSegment.RemotePieces[:i], remoteSegment.RemotePieces[i+1:]...)
```

Comment:
I am not sure if we should do it this way since we are iterating over `remoteSegment.RemotePieces` while we are trying to modify it. Maybe it would be better to have a piece list called `goodPieces` that starts out empty, then while we iterate, if the piece is not failed, we append it to `goodPieces`. After the loop ends, we can say `remoteSegment.RemotePieces = goodPieces` and not have to worry about weird slice stuff inside the loop

Reply:
fixing